src/Symfony/Component/Filesystem/Tests/Fixtures/MockStream/MockStream.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace MockStream;
+
+/**
+ * Mock stream class to be used with stream_wrapper_register.
+ * stream_wrapper_register('mock', 'MockStream\MockStream').
+ */
+class MockStream
+{
+    private $strOverloaded;
+    private $content;
+    private $position;
+    private $atime;
+    private $mtime;
+    private $ctime;
+    private $path;
+
+    /**
+     * Opens file or URL.
+     *
+     * @param string $path        Specifies the URL that was passed to the original function
+     * @param string $mode        The mode used to open the file, as detailed for fopen()
+     * @param int    $options     Holds additional flags set by the streams API
+     * @param string $opened_path If the path is opened successfully, and STREAM_USE_PATH is set in options,
+     *                            opened_path should be set to the full path of the file/resource that was actually
+     *                            opened
+     *
+     * @return bool
+     */
+    public function stream_open($path, $mode, $options, &$opened_path)
+    {
+        // Is mbstring.func_overload applied to string functions (bit 2 set)
+        $this->strOverloaded = (bool) (ini_get('mbstring.func_overload') & (1 << 2));
+        $this->path = $path;
+        $this->content = '';
+        $this->position = 0;
+        $this->atime = 0;
+        $this->mtime = 0;
+
+        return true;
+    }
+
+    /**
+     * Read from stream.
+     *
+     * @param int $count How many bytes of data from the current position should be returned
+     *
+     * @return string The data
+     */
+    public function stream_read($count)
+    {
+        $ret = $this->substr($this->content, $this->position, $count);
```

Comment:
for what's the mock is used for, almost all implementations should be left empty. It's dead code currently.